libkatana_python_native/src/PropertyGraph.cpp

```diff
@@ -0,0 +1,742 @@
+#define NPY_NO_DEPRECATED_API NPY_1_11_API_VERSION
+
+#include <numpy/ndarrayobject.h>
+
+// Must come after numpy/ndarrayobject.h since that is required, but
+// not included by arrow/python headers.
+#include <arrow/python/numpy_convert.h>
+#include <arrow/python/numpy_to_arrow.h>
+#include <arrow/python/python_to_arrow.h>
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "katana/python/Conventions.h"
+#include "katana/python/CythonIntegration.h"
+#include "katana/python/EntityTypeManagerPython.h"
+#include "katana/python/ErrorHandling.h"
+#include "katana/python/NumbaSupport.h"
+#include "katana/python/PropertyGraphPython.h"
+#include "katana/python/PythonModuleInitializers.h"
+#include "katana/python/TemplateSupport.h"
+#include "katana/python/TypeBans.h"
+
+namespace py = pybind11;
+
+katana::Result<std::shared_ptr<arrow::Table>>
+katana::python::PythonArgumentsToTable(
```

Comment:
This is a really cool method and gives me hope that we can more naturally create RDGs from dataframes without having to go through a round-about of calling CSV import.

Reply:
:-) Yea. You can already write things like `g.add_node_property(pandas_df)`. I think it's tested in the pytests.